gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/longrunning/stub/HttpJsonOperationsStub.java

```diff
@@ -223,21 +230,58 @@ public static final HttpJsonOperationsStub create(
       TypeRegistry typeRegistry)
       throws IOException {
     return new HttpJsonOperationsStub(
-        OperationsStubSettings.newBuilder().build(), clientContext, callableFactory, typeRegistry);
+        OperationsStubSettings.newBuilder().build(),
+        clientContext,
+        callableFactory,
+        typeRegistry,
+        new HashMap<>());
+  }
+
+  // We choose to only override these two constructors
+  // The generator can generate these two combinations with custom HttpRules
+  public static final HttpJsonOperationsStub create(
```

Comment:
Calling one seems can be replaced by calling the one below with `TypeRegistry.getEmptyTypeRegistry()`. If there is no other major benefit, I would remove this constructor.

Reply:
I've kept this one in because it is currently possible for the generator to generate this `create()` function with only two parameters. Though, searching in the codebase only suggests that three parameters has only ever been generated, I wasn't sure if that would always be the case going forward.

Reply:
On a second look, I can probably just remove the other constructor and add the param when calling the constructor.

Reply:
Yes, as the protected constructor could be called by the subclass's constructor as well.

Reply:
Ok in that case, I'll keep these changes as is. Given the generator is able to call both:
- ClientContext clientContext, HttpJsonStubCallableFactory callableFactory, Map<String, HttpRule> customHttpBindings
- ClientContext clientContext, HttpJsonStubCallableFactory callableFactory, TypeRegistry typeRegistry, Map<String, HttpRule> customHttpBindings

Reply:
Maybe I'm missing something, theoretically we only need to expose one public constructor that takes the additional map, why do we have two new public constructors and two new protected constructors? Is it just for future convenience?

Reply:
The generator currently can create two variants of `HttpJsonOperationStub.create()` at https://github.com/googleapis/gapic-generator-java/blob/b43d9fc7f25800861847d4324298f1e16da5a508/gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java#L1086-L1090
1. HttpJsonOperationStub.create(clientContext, callableFactory)
2. HttpJsonOperationStub.create(clientContext, callableFactory, typeRegistry)

Though, it seems that every version in google-cloud-java has the 2nd option and I'm not sure if there is a use case for the 1st option. I chose to implement both since I'm not 100% sure that the 1st option would never be called. Now there are 4 possibilities (and realistically probably only 2 used -- the 2nd and 4th):
1. HttpJsonOperationStub.create(clientContext, callableFactory)
2. HttpJsonOperationStub.create(clientContext, callableFactory, typeRegistry)
3. HttpJsonOperationStub.create(clientContext, callableFactory, ImmutableMap.of(...))
4. HttpJsonOperationStub.create(clientContext, callableFactory, typeRegistry, ImmutableMap.of(...))

I think I can probably make the newly added constructor private though.

Reply:
I agree. Also if there is no usage of `HttpJsonOperationsStub.create(clientContext, callableFactory)`, I don't see why we have to keep maintaining `HttpJsonOperationsStub.create(clientContext, callableFactory)`, we can deprecate it and refactor the code you mentioned above to always use `HttpJsonOperationStub.create(clientContext, callableFactory, typeRegistry, ImmutableMap.of(...))`. It could be a separate PR though.

Reply:
Sounds good. I'll remove `HttpJsonOperationsStub create(ClientContext clientContext, HttpJsonStubCallableFactory callableFactory, Map<String, HttpRule> customHttpBindings)` any anything else not needed. 

Since `HttpJsonOperationsStub.create(clientContext, callableFactory)` is already public at: https://github.com/googleapis/gapic-generator-java/blob/0ccf457a774bd3dbe3b101d82222bdc39f6e5b0e/gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/longrunning/stub/HttpJsonOperationsStub.java#L211-L218, I'll just leave it as is.